topics/icrc_X_multiple_canister_call.md

```diff
@@ -0,0 +1,237 @@
+# ICRC-X: Call batch canisters
+
+[![Status Badge](https://img.shields.io/badge/STATUS-DRAFT-ffcc00.svg)](#)
+[![Extension Badge](https://img.shields.io/badge/Extends-ICRC--25-ffcc222.svg)](./icrc_25_signer_interaction_standard.md)
+[![Extension Badge](https://img.shields.io/badge/Extends-ICRC--49-ffcc222.svg)](./icrc_49_call_canister.md)
+
+<!-- TOC -->
+
+- [ICRC-X: Call batch canisters](#icrc-x-call-batch-canisters)
+  - [Summary](#summary)
+  - [Processing](#processing)
+    - [Examples](#examples)
+      - [Successful Batch Call](#successful-batch-call)
+      - [Batch Call with Error](#batch-call-with-error)
+
+## Summary
+
+This Method can be used by the relying party to request batch call to 3rd party canister executed by the signer using the requested identity. In order to prevent misuse of this method all `icrcX_call_batch_canisters` requests are subject to user approval.
+
+There is two main parameter for this standard
+
+1. `mode` : The mode to execute transaction in sequence or in parallel
```

Comment:
Instead of picking between a mode, wouldn't it be more flexible to combine modes?

For example a user might want to ask 2 approvals in parallel before making a call to a swap. To do something like that I'd either nest requests inside requests of for even more flexibility, add a dependency array to each request that indicates which requests must be finished before the request is executed.

An example of optional dependency arrays would look something like this.

```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "icrcX_batch_call_canisters",
  "params": {
    "sender": "b7gqo-ulk5n-2kpo7-oalt7-p2kyl-o4j5l-kiuwo-eeybr-dab4l-ur6up-pqe",
    "requests": [
      {
        "id": 1,
        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
        "method": "approve",
        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
      },
      {
        "id": 2,
        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
        "method": "approve",
        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
      },
      {
        "id": 3,
        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
        "method": "swap",
        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
        "waitFor": [1, 2],
      },
      {
        "id": 4,
        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
        "method": "swap",
        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
        "waitFor": [3],
      }
    ]
  }
}
```

Requests would execute in this example: `1 & 2 -> 3 -> 4`

Reply:
but how do we know the dependencies 1 or 2 are success or not in order to execute 3

Reply:
So from the question in discord 

```
If we really want to implement sequence, we could ask the relying party to pass the expected success response, such as 'ok' (i..e  ICRC 1 or 2 ) or a non-null blockid, for each transaction as parameters to signer, and signer continues to execute the next tx only if the success code is received.
```

Can this possible to do? 
Based on your answer is yes for signer parsing data
But how about relying party passing `expected success response` to the signer like regex or maybe a full payload